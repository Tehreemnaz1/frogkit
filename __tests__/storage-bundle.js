import { mount, shallow } from 'vue-test-utils'
import Vue from 'vue'
import StorageBundle from '@/components/storage-bundle'

const propsData = {
  storage: {
    title: 'Speichererweiterung',
    for: 'Speichererweiterung für:',
    devices: [{value: '1', label: 'iphone 8'}, {value: '2', label: 'HW Cam DEV'}, {value: '3', label: 'SM-T825'}],
    history: 'Video-Speicher der jeweils letzten:',
    upgrades: [
      {value: '24h', label: '24 Std. 2,65€ / mtl.'},
      {value: '7d', label: '7 Tage 5,35€ / mtl.'},
      {value: '30d', label: '30 Tage 13,45€ / mtl.'}
    ],
    price: {
      amount: '5.35',
      currency: 'EUR',
      prefix: 'nur',
      suffix: 'im Monat'
    },
    action: 'Video-Speicher kaufen'
  },
  selectedDevice: {value: '1', label: 'iphone 8'},
  selectedUpgrade: {value: '24h', label: '24 Std. 2,65€ / mtl.'},
  active: true
}


describe('Storage Bundle component', () => {
  it('should render component and match snapshot', () => {
    // Note: a hack to get appropriate snapshot
    StorageBundle.data = () => { return {show: true} }
    const wrapper = mount(StorageBundle, {propsData})
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit upgrade event', () => {
    const wrapper = mount(StorageBundle, {propsData})
    const stub = jest.fn()
    wrapper.vm.$on('upgrade', stub)
    wrapper.findAll('button').at(0).trigger('click')
    expect(stub).toBeCalled()
  })
  it('should emit select event', () => {
    const wrapper = mount(StorageBundle, {propsData})
    const stub = jest.fn()
    wrapper.vm.$on('select', stub)
    wrapper.findAll('select').at(0).trigger('input')
    expect(stub).toBeCalled()
  })
  it('should emit change event', () => {
    const wrapper = mount(StorageBundle, {propsData})
    const stub = jest.fn()
    wrapper.vm.$on('change', stub)
    wrapper.findAll('radio').at(0).trigger('change')
    expect(stub).toBeCalled()
  })
})
