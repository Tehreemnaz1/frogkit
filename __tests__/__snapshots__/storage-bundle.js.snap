// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storage Bundle component should render component and match snapshot 1`] = `
<flex align="center" column="" class="fk-storage-bundle fk-storage-bundle__container">
    <title size="xs" color="primary" center="" class="fk-storage-bundle__title">Speichererweiterung</title>
    <div class="fk-storage-bundle__icon"></div>
    <p class="fk-storage-bundle__for">Speichererweiterung für:</p> <select options="[object Object],[object Object],[object Object]" value="1" class="fk-storage-bundle__devices"></select>
    <p class="fk-storage-bundle__history">Video-Speicher der jeweils letzten:</p>
    <div>
        <radio value="24h" checked="checked" class="fk-storage-bundle__period">
            <p class="fk-storage-bundle__period__label">24 Std. 2,65€ / mtl.<sup>2</sup></p>
        </radio>
        <radio value="7d" checked="checked" class="fk-storage-bundle__period">
            <p class="fk-storage-bundle__period__label">7 Tage 5,35€ / mtl.<sup>2</sup></p>
        </radio>
        <radio value="30d" checked="checked" class="fk-storage-bundle__period">
            <p class="fk-storage-bundle__period__label">30 Tage 13,45€ / mtl.<sup>2</sup></p>
        </radio>
    </div>
    <flex align="center" column="" class="fk-storage-bundle__info">
        <p class="m-b-15">
            <pricetag value="5.35" code="EUR" class="fk-storage-bundle__price">
                <p slot="prefix">nur</p>
                <p slot="suffix">im Monat<sup>2</sup></p>
            </pricetag>
        </p> <button color="secondary" size="big" block="" class="fk-storage-bundle__action">Video-Speicher kaufen</button></flex>
</flex>
`;
